### ISSUE TITLE
Authorization code already redeemed.

### ISSUE THREAD

**Issue by wyber** [AzureAD/microsoft-authentication-library-for-android#638]
MSAL returned the following exception:
```
Caused by: AADSTS54005: OAuth2 Authorization code was already redeemed, please retry with a new valid code or use an existing refresh token.
Trace ID: 4e6ca37f-5931-469d-9307-da01633d2500
Correlation ID: 014b01f7-7fda-43f4-b7de-1ace5fcba67d
Timestamp: 2019-05-23 14:21:13Z
com.microsoft.identity.client.internal.controllers.MsalExceptionAdapter.msalExceptionFromBaseException(MsalExceptionAdapter.java:62)
com.microsoft.identity.client.PublicClientApplication$3.onError(PublicClientApplication.java:1233)
com.microsoft.identity.common.internal.controllers.ApiDispatcher$1$4.run(ApiDispatcher.java:126)
android.os.Handler.handleCallback(Handler.java:790)
android.os.Handler.dispatchMessage(Handler.java:99)
android.os.Looper.loop(Looper.java:164)
android.app.ActivityThread.main(ActivityThread.java:6556)
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

Based on this answer: https://social.msdn.microsoft.com/Forums/en-US/4192e141-309a-4dd6-a5c9-f1a8ce32f4ca/aadsts54005-oauth2-authorization-code-was-already-redeemed?forum=WindowsAzureAD

This seems like a bug. It would require MSAL to be using the same authorization code twice.

**Comment by iambmelt** [comment]
Thanks @wyber do you have repro steps?

**Comment by wyber** [comment]
This stack is from telemetry, so no, unfortunately I do not. The code path this goes down is calling one of:
```
    public void acquireToken(@NonNull final Activity activity,
                             @NonNull final String[] scopes,
                             @Nullable final IAccount account,
                             @NonNull final UiBehavior uiBehavior,
                             @Nullable final List<Pair<String, String>> extraQueryParameters,
                             @Nullable final String[] extraScopesToConsent,
                             @Nullable final String authority,
                             @NonNull final AuthenticationCallback callback)
```
or
```
    public void acquireToken(@NonNull final Activity activity,
                             @NonNull final String[] scopes,
                             @Nullable final String loginHint,
                             @Nullable final UiBehavior uiBehavior,
                             @Nullable final List<Pair<String, String>> extraQueryParameters,
                             @Nullable final String[] extraScopesToConsent,
                             @Nullable final String authority,
                             @NonNull final AuthenticationCallback callback)
```

**Comment by iambmelt** [comment]
This should now be fixed in our `1.0.0` release.

If anyone else has this issue, please open a new issue and include logs/repro steps.

Thanks
